addons/mail/models/discuss/discuss_call_history.py

```diff
@@ -0,0 +1,27 @@
+from odoo import api, models, fields
+from odoo.addons.mail.tools.discuss import Store
+
+
+class DiscussCallHistory(models.Model):
+    _name = "discuss.call.history"
+    _description = "Keep the call history"
+
+    channel_id = fields.Many2one("discuss.channel", index=True, ondelete="cascade")
+    duration = fields.Float(compute="_compute_duration", store=True)
```

Comment:
Is computing end - start in reporting so complex that we have to store an extra field?
Do we ever possibly need to group, sort or filter by duration in other views?

Reply:
It's also used in `notification_message`. I don't want to compute it in two places.